src/components/shared/Modal.js

```diff
@@ -35,6 +34,15 @@ export default class Modal extends Component {
     shortcuts.on(L10N.getStr(this.props.shortcut), this.props.handleOpen);
   }
 
+  componentDidUpdate() {
+    const shortcuts = this.context.shortcuts;
+    if (this.props.enabled) {
```

Comment:
hmm, this worries me because we might be doing too many `on` calls.

Should we add a check in here like `!props.enabled && nextProps.enabled`?